packages/backend-app-api/src/wiring/types.ts

```diff
@@ -45,9 +45,16 @@ export interface CreateSpecializedBackendOptions {
   services: (ServiceFactory | (() => ServiceFactory))[];
 }
 
-export type ServiceHolder = {
+export interface ServiceHolder {
   get<T>(api: ServiceRef<T>, pluginId: string): Promise<T> | undefined;
-};
+}
+
+/**
+ * @internal
+ */
+export interface EnumerableServiceHolder extends ServiceHolder {
+  getServiceRefs(): ServiceRef<unknown>[];
```

Comment:
Just out of curiosity: Why is the generic type variable of the ServiceRef `unknown`? Couldn't it be smth. similar to the above like:

```suggestion
  getServiceRefs<T>(): ServiceRef<T>[];
```

Or does that not provide no advantages? :/

Reply:
We actually don't know what ServiceRefs are returned here so `unknown` is correct. For `get<T>` we can resolve the type from the serviceRef passed in :)